Library/Dependencies/napi/source/js_native_api_chakra.cc

```diff
@@ -1,337 +1,124 @@
-  do {                                                                  \
-    JsErrorCode err = (expr);                                           \
-    if (err != JsNoError) return napi_set_last_error(err);              \
-  } while (0)
-
-#define CHECK_JSRT_EXPECTED(expr, expected)                             \
-  do {                                                                  \
-    JsErrorCode err = (expr);                                           \
-    if (err == JsErrorInvalidArgument)                                  \
-      return napi_set_last_error(expected);                             \
-    if (err != JsNoError) return napi_set_last_error(err);              \
-  } while (0)
-
-// This does not call napi_set_last_error because the expression
-// is assumed to be a NAPI function call that already did.
-#define CHECK_NAPI(expr)                                                \
-  do {                                                                  \
-    napi_status status = (expr);                                        \
-    if (status != napi_ok) return status;                               \
-  } while (0)
-
-
-// utf8 multibyte codepoint start check
-#define UTF8_MULTIBYTE_START(c) (((c) & 0xC0) == 0xC0)
-
-#define STR_AND_LENGTH(str) str, sizeof(str) - 1
-
-static napi_status napi_set_last_error(napi_status error_code,
-                                       uint32_t engine_error_code = 0,
-                                       void* engine_reserved = nullptr);
-static napi_status napi_set_last_error(JsErrorCode jsError,
-                                       void* engine_reserved = nullptr);
-static void napi_clear_last_error();
-
-// Callback Info struct as per JSRT native function.
-struct CallbackInfo {
-  napi_value newTarget;
-  napi_value thisArg;
-  napi_value* argv;
-  void* data;
-  uint16_t argc;
-  bool isConstructCall;
-};
 
-// $HACK for ChakraCore
 namespace {
-std::wstring NarrowToWide(std::string_view value)
-{
-    int requiredSize = ::MultiByteToWideChar(CP_UTF8, 0, value.data(), static_cast<int>(value.size()), nullptr, 0);
-    std::wstring wstr(requiredSize, 0);
-    ::MultiByteToWideChar(CP_UTF8, 0, value.data(), static_cast<int>(value.size()), &wstr[0], requiredSize);
-    return std::move(wstr);
-}
-
-STDAPI_(JsErrorCode) JsCreateString(
-    _In_ const char* content,
-    _In_ size_t length,
-    _Out_ JsValueRef* value)
-{
-    auto str = length == NAPI_AUTO_LENGTH ? NarrowToWide({ content }) : NarrowToWide({ content, length });
-    return JsPointerToString(str.data(), str.size(), value);
-}
-
-STDAPI_(JsErrorCode) JsCopyString(
-    _In_ JsValueRef value,
-    _Out_opt_ char* buffer,
-    _In_ size_t bufferSize,
-    _Out_opt_ size_t* length)
-{
-    const wchar_t* stringValue;
-    size_t stringLength;
-    auto errorCode = JsStringToPointer(value, &stringValue, &stringLength);
+constexpr unsigned int CP_LATIN1 = 28591;
```

Comment:
nit: Use the `UINT` typedef, just to guarantee it matches `MultiByteToWideChar`? I don't know if `UINT` can never not be the same as `unsigned int`, though I'm sure it would matter much in either case.